command/agent_test.go

```diff
@@ -397,6 +396,9 @@ listener "tcp" {
 	defer func() {
 		cmd.ShutdownCh <- struct{}{}
 		wg.Wait()
+		if output != "" {
+			t.Fatalf("got a non-zero exit status: %s", output)
```

Comment:
This seems incorrect to me. If we write to stdout, it doesn't mean we got a non-zero exit status

Reply:
The output is only constructed if we get a non-zero exit status. What message do you want to see here?

Reply:
The output is only constructed if we get a non-zero exit status, but we're not actually checking the exit status, we're checking stdout/stderr instead. 

Previously, we fail if we get a non-zero return code

Now, we fail if we get a non-zero return code AND we have output in stdout/stderr

It might make sense to track the return code, instead of the output?

Reply:
So, I think our best option here is to record the `code`, not the output, then have the check that tracks the return code in this defer. It might be a still good idea to log stdout and stderr, in these cases, as it helps with debugging.

Thanks for tackling this, by the way! Super good catch :)